pkg/upgrade/upgrade.go

```diff
@@ -275,3 +276,30 @@ func upgradeToVer3(ctx context.Context, cli *clientv3.Client) error {
 func upgradeToVer4(cli *clientv3.Client, uctx Context) error {
 	return nil
 }
+
+// upgradeToVer5 does upgrade operations from Ver4 to Ver5 (v2.0.7) to update online-ddl from online-ddl-scheme in configure.
+func upgradeToVer5(cli *clientv3.Client, uctx Context) error {
+	upgradeTaskName := "upgradeToVer5"
+	logger := log.L().WithFields(zap.String("task", upgradeTaskName))
+
+	if uctx.SubTaskConfigs == nil {
+		logger.Info("no downstream DB, skipping")
+		return nil
+	}
+
+	for task, m := range uctx.SubTaskConfigs {
+		for sourceID, subCfg := range m {
+			subCfg.AdjustOnlineDDL()
+			value, err := subCfg.Toml()
+			if err != nil {
+				return err
+			}
+			key := common.UpstreamSubTaskKeyAdapter.Encode(sourceID, task)
```

Comment:
I suddenlt realized that the upgrading actions can't make sure that DM-worker will see the new config. If DM-worker finish upgrading too quickly, at the time of its starting up, DM-master has not run here so task in etcd is not changed.

Maybe we should change the upgrading logic in TiUP 🤔

Reply:
Some possible solutions:
1. Let worker do adjust job again.
2. If a master finds that its own version info is greater than version Info in etcd, transfer leader to itself immediately and boostrap before this dm-master provides service.
I think we can do this in another PR. It's hard and inappropriate for TiUP to handle DM's own upgrade logic.

Reply:
I have checked the logic in dm-worker. dm-worker will adjust subtask config again when dm-worker reads configuration from etcd. So maybe we don't have such problem?